api/tasks.py

```diff
@@ -4,25 +4,38 @@
 import wordpress
 from wordpress.user import WPUser
 from lazona_connector.celery import app
+from api.models import Shipment, ShipmentStatus
 
 
 @app.task
 def create_delivery(order):
     deliveries_by_vendor = Client(order).create_delivery()
     for delivery in deliveries_by_vendor:
+        label_url = ""
         if delivery._is_errored():
+            delivery_status = ShipmentStatus.ERROR_FROM_BODY
             FailedDeliveryMail(
                 order_id=delivery.order_id,
                 error_returned=delivery.data.get("mensaje"),
                 req_body=delivery.req_body
             ).send()
         else:
+            delivery_status = ShipmentStatus.LABEL_SENT
+            label_url = delivery.print_pdf()
             SuccessDeliveryMail(
-                pdf_path=delivery.print_pdf(),
+                pdf_path=label_url,
                 recipient=delivery.vendor.email,
                 order_id=delivery.order_id
             ).send()
 
+        Shipment(
+            delivery_id=delivery.data.get("codBarras", ""),
```

Comment:
However, I'm not following why we have to violate the delivery object's encapsulation and get to its raw data through `data()`. IMO this beats the purpose of that class and it'd be much better to implement a getter for the barcode. So we can do `delivery.barcode` :ok_hand: 

This would give us the chance to deal with these details of whether or not we raise a KeyError or provide a default value within `Delivery` itself, uncoupling consumers from that, like this `api/tasks.py`.

Reply:
Fixed at 7a276ab16904e9c16c2e45c4c73a5424b96f65e6